js/geo/transform.js

```diff
@@ -375,7 +394,8 @@ Transform.prototype = {
                 y2 !== undefined ? y2 : this.y));
         }
 
-        this._constraining = false;
+        this._unmodified = this._constraining.unmodified;
+        this._constraining = undefined;
```

Comment:
I store it there so I can restore `this._unmodified` back to its original value after `_constrain()` is finished.

`_constrain()` should never change the modified state of the transform, even though it needs to be able to mess with with `this.zoom` and `this.center`.